app/src/main/java/net/gsantner/markor/util/AppSettings.java

```diff
@@ -543,6 +602,15 @@ public ArrayList<File> getFavouriteFiles() {
         return list;
     }
 
+    public ArrayList<File> getFavouriteFilesAll() {
```

Comment:
add a bool parameter to the existing favourites  -->

`if (checkExists && (exist || virtual))`

Reply:
This function performs for the purpose of noticing those files which are still recorded but is actually deleted already.

Reply:
yes, but still: It's the same like the normal `getFavouriteFiles(`, except the `if` condition.

Reply:
I merged two into one and deleted the `if` statement, because I think if code runs correctly, when the function is called in other situations it shouldn't contain file paths which does not meet the constraints. Still, if adding a parameter would be better, that also seems pretty good to me.

Reply:
> I think if code runs correctly, when the function is called in other situations it shouldn't contain file paths which does not meet the constraints.


NO! You are wrong.   Users can also delete file outside of Markor. Even automatically, some synchronisation software could delete the file. So just thinking `ok we remove it from the list when we delete it inside Markor` is not enough.

Reply:
> I think if code runs correctly, when the function is called in other situations it shouldn't contain file paths which does not meet the constraints.


NO! You are wrong.   Users can also delete file outside of Markor. Even automatically, some synchronisation software could delete the file. So just thinking `ok we remove it from the list when we delete it inside Markor` is not enough. There MUST be validation check for file existing - at time of loading the favourites. This is what the if condition is about.

Reply:
Oh, I see. I understand it now, I'll do the updates.

Reply:
There are still a few places unchanged, I'll solve them a bit later.

Reply:
> There are still a few places unchanged, I'll solve them a bit later.

Now the files are up to date and several tests are done.